zerver/actions/message_edit.py

```diff
@@ -398,13 +399,16 @@ def do_update_embedded_data(
         "rendering_only": True,
     }
 
+    users_to_notify = event_recipient_ids_for_action_on_messages([message])
```

Comment:
Does it make sense to query that first, before doing `ums = UserMessage.objects.filter(message=message.id)`? 

Maybe it's unavoidable, but it seems like we should at least look if it's easy to avoid querying `UserMessage` twice in the same code path.

Reply:
I guess relatedly, is it correct to have `exclude_long_term_idle_users=True` in this code path? 

(The risk would be that we do use message events to determine if we need to send an email/push notification for a user even if they were long-term idle, and I don't remember off the top of my head if the message-edit code path needs to interact with that in some way)

Reply:
I think we can't avoid querying `UserMessage` twice here because first we need to update flags for all objects and then `event_recipient_ids_for_action_on_messages` involves applying various filters on the UserMessage queryset.

One approach would be to fetch everything in memory once (usermessage objects with `user_profile__role` and `user_profile__long_term_idle` ) and pass it to `event_recipient_ids_for_action_on_messages` to perform the filtering in python. -- But I think this is inefficient to do in python (memory and speed wise) because of the large number of usermessage objects ?